server/mscalendar/availability.go

```diff
@@ -21,78 +22,102 @@ const (
 	StatusSyncJobInterval           = 5 * time.Minute
 	upcomingEventNotificationTime   = 10 * time.Minute
 	upcomingEventNotificationWindow = (StatusSyncJobInterval * 11) / 10 // 110% of the interval
+	logTruncateMsg                  = "We've truncated the logs due to too many messages"
+	logTruncateLimit                = 5
 )
 
+type StatusSyncJobSummary struct {
+	NumberOfUsersFailedStatusChanged int
+	NumberOfUsersStatusChanged       int
+	NumberOfUsersProcessed           int
+}
+
 type Availability interface {
 	GetCalendarViews(users []*store.User) ([]*remote.ViewCalendarResponse, error)
-	Sync(mattermostUserID string) (string, error)
-	SyncAll() (string, error)
+	Sync(mattermostUserID string) (string, *StatusSyncJobSummary, error)
+	SyncAll() (string, *StatusSyncJobSummary, error)
 }
 
-func (m *mscalendar) Sync(mattermostUserID string) (string, error) {
+func (m *mscalendar) Sync(mattermostUserID string) (string, *StatusSyncJobSummary, error) {
 	user, err := m.Store.LoadUserFromIndex(mattermostUserID)
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 
 	return m.syncUsers(store.UserIndex{user})
 }
 
-func (m *mscalendar) SyncAll() (string, error) {
+func (m *mscalendar) SyncAll() (string, *StatusSyncJobSummary, error) {
 	err := m.Filter(withSuperuserClient)
 	if err != nil {
-		return "", err
+		return "", &StatusSyncJobSummary{}, errors.Wrap(err, "not able to filter the super user client")
```

Comment:
Nit pick: In general, functions that return a non-nil error should have the other fields set to `nil`.

Reply:
@hanzei  Actually here I am not returning nil because, in its parent functions, I am using an attribute of this object, and if I return nil from here that could give panic.